internal/services/firewall/firewall_data_source.go

```diff
@@ -151,6 +152,26 @@ func firewallDataSource() *pluginsdk.Resource {
 				},
 			},
 
+			"autoscale_configuration": {
+				Type:        pluginsdk.TypeSet,
+				Description: "Properties to provide a custom autoscale configuration to this azure firewall.",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"min_capacity": {
+							Type:        pluginsdk.TypeInt,
+							Description: "The minimum number of capacity units for this azure firewall. Use null to reset the value to the service default.",
+							Computed:    true,
+						},
+						"max_capacity": {
+							Type:        pluginsdk.TypeInt,
+							Description: "The maximum number of capacity units for this azure firewall. Use null to reset the value to the service default.",
```

Comment:
The description can be removed as this provider doesn't use it to generate the document.